components/ai_chat/core/common/proto_conversion_unittest.cc

```diff
@@ -0,0 +1,268 @@
+  mojom_event->sources.push_back(std::move(invalid_url_source));
+
+  auto invalid_favicon_source = mojom::WebSource::New(
+      "Invalid Favicon Source", GURL("https://example.com/valid"),
+      GURL("invalid-favicon-url"));
+  mojom_event->sources.push_back(std::move(invalid_favicon_source));
+
+  // Serialize to proto
+  store::WebSourcesEventProto proto_event;
+  SerializeWebSourcesEvent(mojom_event, &proto_event);
+
+  // Only the valid source should be serialized
+  EXPECT_EQ(proto_event.sources_size(), 1);
+  EXPECT_EQ(proto_event.sources(0).title(), "Valid Source");
+}
+
+TEST(ProtoConversionTest, DeserializeWebSourcesEvent_InvalidUrls) {
+  // Create proto WebSourcesEvent with invalid URLs
+  store::WebSourcesEventProto proto_event;
+
+  auto* valid_source = proto_event.add_sources();
+  valid_source->set_title("Valid Source");
+  valid_source->set_url("https://example.com/valid");
+  valid_source->set_favicon_url("https://example.com/valid.ico");
+
+  auto* invalid_url_source = proto_event.add_sources();
+  invalid_url_source->set_title("Invalid URL Source");
+  invalid_url_source->set_url("invalid-url");
+  invalid_url_source->set_favicon_url("https://example.com/valid.ico");
+
+  auto* invalid_favicon_source = proto_event.add_sources();
+  invalid_favicon_source->set_title("Invalid Favicon Source");
+  invalid_favicon_source->set_url("https://example.com/valid");
+  invalid_favicon_source->set_favicon_url("invalid-favicon-url");
+
+  // Deserialize to mojom
+  auto mojom_event = DeserializeWebSourcesEvent(proto_event);
+
+  // Only the valid source should be deserialized
+  EXPECT_EQ(mojom_event->sources.size(), 1u);
+  EXPECT_EQ(mojom_event->sources[0]->title, "Valid Source");
+}
+
+TEST(ProtoConversionTest, SerializeDeserializeWebSourcesEvent_EmptySources) {
+  // Create empty mojom WebSourcesEvent
+  auto mojom_event = mojom::WebSourcesEvent::New();
+
+  // Serialize to proto
+  store::WebSourcesEventProto proto_event;
+  SerializeWebSourcesEvent(mojom_event, &proto_event);
+
+  // Verify empty proto
+  EXPECT_EQ(proto_event.sources_size(), 0);
+
+  // Deserialize back to mojom
+  auto deserialized_event = DeserializeWebSourcesEvent(proto_event);
+
+  // Verify empty mojom
+  EXPECT_EQ(deserialized_event->sources.size(), 0u);
+}
+
+// Tests for ToolUseEvent conversion functions
+
+TEST(ProtoConversionTest, SerializeDeserializeToolUseEvent_ValidData) {
+  const std::string kArgumentsJson = R"({
+        "param1": "value1",
+        "param2": 42,
+        "param3": true,
+        "nested": {
+          "nestedparam1": "value2",
+          "nestedparam2": 43
+        }
+      })";
+  // Create mojom ToolUseEvent
+  auto mojom_event = mojom::ToolUseEvent::New(
+      "test_tool", "tool_id_123", kArgumentsJson, base::Value::Dict(),
+      std::vector<mojom::ContentBlockPtr>());
+
+  // Add different parsed arguments to verify that they are not the source
+  // or truth - argments_json should be.
```

Comment:
s/source or truth/source of truth/
s/argments_json/arguments_json/

and can we pass this through dict when we create mojom::ToolUseEvent?